File: test-suite-python/src/test/python/micronaut/servlet/docs/jetty/JettyServerCustomizer.py
# tag::imports[]
from jakarta.inject import Singleton
from micronaut.context.event import BeanCreatedEvent, BeanCreatedEventListener
from org.eclipse.jetty.server import Server
# end::imports[]
from micronaut.context.annotation import Requires


@Requires(property="spec.name", value="JettyServerCustomizerTest")
# tag::class[]
@Singleton
class JettyServerCustomizer(BeanCreatedEventListener[Server]):

    def onCreated(self, event: BeanCreatedEvent[Server]) -> Server:
        jetty_server = event.getBean()
        # perform customizations, for example:
        jetty_server.setStopTimeout(5000)
        return jetty_server
# end::class[]

File: test-suite/src/test/java/io/micronaut/servlet/docs/jetty/JettyServerCustomizer.java
package io.micronaut.servlet.docs.jetty;

// tag::imports[]
import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;
import org.eclipse.jetty.server.Server;
import jakarta.inject.Singleton;
// end::imports[]
import io.micronaut.context.annotation.Requires;

@Requires(property = "spec.name", value = "JettyServerCustomizerTest")
// tag::class[]
@Singleton
public class JettyServerCustomizer implements BeanCreatedEventListener<Server> {
    @Override
    public Server onCreated(BeanCreatedEvent<Server> event) {
        Server jettyServer = event.getBean();
        // perform customizations, for example:
        jettyServer.setStopTimeout(5000);
        return jettyServer;
    }
}
// end::class[]
